### PR TITLE
Allow user data path to be specified by command line

### PR DESCRIPTION
**Pull request by IntelOrca**
See #2182

Added ```--user-data-path``` command argument so that you can specify where your user directory is allowing OpenRCT2 to be played completely off an external drive.

Also changed default path under linux to be ~/.config/OpenRCT2.

@janisozaur can you test this on linux please?

### CODE REVIEW

File: src/platform/linux.c [hunk 1]
```diff
@@ -554,8 +556,25 @@ wchar_t *regular_to_wchar(const char* src)
...
+ *   - /home/[uid]/.config/OpenRCT2
+ */
+void platform_resolve_user_data_path()
 {
+	if (gCustomUserDataPath[0] != 0) {
+		safe_strncpy(_userDataDirectoryPath, gCustomUserDataPath, sizeof(_userDataDirectoryPath));
+
+		// Ensure path ends with separator
+		int len = strlen(_userDataDirectoryPath);
+		if (_userDataDirectoryPath[len - 1] != separator[0]) {
```
Comment (janisozaur) [inline src/platform/linux.c]: `separator` is undefined here

File: src/platform/linux.c [hunk 2]
```diff
@@ -567,16 +586,32 @@ void platform_get_user_directory(utf8 *outPath, const utf8 *subDirectory)
...
-			log_error("Couldn't find user home directory");
+			log_fatal("Couldn't find user data directory");
+			exit(-1);
 			return;
 		}
 	}
 	char separator[2] = { platform_get_path_separator(), 0 };
 	strncat(buffer, homedir, MAX_PATH);
 	strncat(buffer, separator, MAX_PATH);
+	strncat(buffer, ".config", MAX_PATH);
```
Comment (janisozaur) [inline src/platform/linux.c]: `XDG_CONFIG_HOME` should already contain `/.config` part, and in it's absence, `$HOME/.config` should be used as a default value.

This line should stay as is, `/.config` should be added in the `getpwuid` block instead. http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html#variables

**Comment by osconnoisseur** [comment]
I haven't tested this yet (I still need to learn how to use Git properly), but platform_resolve_user_data_path() should return an absolute path. GetFullPathName() can convert it on Windows, don't know about Linux. The open/save dialogs are very awkward if you give them a relative path.

**Comment by TurtleShroom** [comment]
Does this matter permit changing the name of the folders in which ORCT2 files, tracks, etc. are stored?

**Pull request merged**